File: javascript/03_injection.js
// A03 - Injection. VULN: command injection via child_process.
const { exec } = require("child_process");

// VULN: A03 - OS command injection
function pingHost(host, cb) {
  exec(`ping -c 1 ${host}`, cb);
}

module.exports = { pingHost };

File: c/03_injection.c
/* A03 - Injection. VULN: command injection via system(). */
#include <stdlib.h>

/* VULN: A03 - OS command injection */
void ping_host(const char *host) {
  char cmd[256];
  snprintf(cmd, sizeof(cmd), "ping -c 1 %s", host);
  system(cmd); /* VULN: unsanitized host */
}
